azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -59,95 +64,152 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
         """
         uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
         self.xml_text = wire_client.fetch_config(uri, wire_client.get_header())
-        xml_doc = parse_doc(self.xml_text)
+        self._xml_doc = parse_doc(self.xml_text)
```

Comment:
there is no need to keep a reference to the xml doc as a member of the goal state instance; it can be passed as argument to the methods that need it

Reply:
Removed the instance member